src/Interpreters/AsynchronousInsertQueue.cpp

```diff
@@ -473,8 +480,27 @@ try
     format->addBuffer(std::move(last_buffer));
     auto insert_query_id = insert_context->getCurrentQueryId();
 
+    auto finish_entries = [&](bool has_data)
+    {
+        for (const auto & entry : data->entries)
+        {
+            if (!entry->isFinished())
+                entry->finish();
+        }
+
+        if (!log_elements.empty())
+        {
+            auto flush_time = has_data ? std::chrono::system_clock::now() : TimePoint{};
+            auto query_id = has_data ? insert_query_id : "";
+            appendElementsToLogSafe(*insert_log, std::move(log_elements), flush_time, query_id, "");
```

Comment:
why flush_time and query_id should be empty if no data?

Reply:
Because the query is not executed in case of no data.

Reply:
Then why log it at all? And if log it - why mess with flush time? We are actually flushing it. And the same for query id - I think it's here to identify which insert query is responsible for this flushing...
I think if we are actually flushing it these info should be preserved with no regards if data is present or not...

Reply:
I agree, let's add this info too.